spatialos-sdk/src/worker/connection.rs

```diff
@@ -260,10 +262,29 @@ impl Connection for WorkerConnection {
 
     fn send_create_entity_request(
         &mut self,
-        _payload: CreateEntityRequest,
-        _timeout_millis: Option<u32>,
+        entity: Entity,
+        entity_id: Option<EntityId>,
+        timeout_millis: Option<u32>,
     ) -> RequestId<CreateEntityRequest> {
-        unimplemented!()
+        let timeout = match timeout_millis {
+            Some(c) => &c,
+            None => ptr::null(),
+        };
+        let entity_id = match entity_id {
+            Some(e) => &e.id,
+            None => ptr::null(),
+        };
+        let id = unsafe {
+            Worker_Connection_SendCreateEntityRequest(
+                self.connection_ptr,
+                entity.component_data.len() as _,
```

Comment:
Yeah, I would build this `component_data` array only when needed, rather than maintaining it in the `Entity` struct. This is the approach we took with the C++/C#/Java SDKs.

Reply:
Okay, if that's the pattern that's worked in the past then I'll get it setup that way here as well 👍